sw/device/tests/BUILD

```diff
@@ -1714,6 +1714,9 @@ opentitan_functest(
 opentitan_functest(
     name = "spi_host_smoketest",
     srcs = ["spi_host_smoketest.c"],
+    cw310 = cw310_params(
```

Comment:
This should be removed if we remove the chip erase from the test.

Reply:
> ## Open questions:
> 1. Should we remove the chip erase test, since it will jump from 21 seconds to 115 seconds?
> 2. Should we randomize the addresses tested to simulate a Wear-leveling and maximize the flash life-span on CI?

For 1 the issue is if we then do any random reads and expect the chip to be in a certain state. If the tests are correct there should be a pattern that is written before a read for checking, otherwise it's a don't care, X state.
2 mins for this particular test is not insignificant when compared to the overall FPGA timeout values...
For 2 - I would suggest yes we should randomise. We probably also need a separate wear-levelling issue for reporting when the SPI flash is getting to end of useful life, especially if we are going to run the tests frequently.

Reply:
1. Among the available flash chips, do we have one that is larger than 16MB and smaller than 128MB?  We could exchange chip-erase for targeted sector-erase (ie: erase only the sectors we're going to use).
2. NOR flash can wear out, but it is much more durable than NAND flash devices.  Typically, SPI NOR eeproms have durability greater than 100000 erase/program cycles (as published in their specs).  Randomization makes deterministic test execution much harder... at a minimum, we'll need to have the test log the addresses selected so failures can be correlated with target addresses.

One additional point on (2): When we get real silicon back and integrate it into the CI, any test automation using the silicon is probably going to include a bootstrap operation, which includes an erase/programming cycle of the chip's internal NOR flash block.  The number of erase/program cycles of the internal flash macro is considered a proprietary secret, but you can assume that it is less than the number of cycles advertised by a purpose-built SPI flash chip.